### ISSUE TITLE
Deprecation: require projects define a configuration file

### ISSUE THREAD

**Issue by agjohnson** [readthedocs/readthedocs.org#10342]
***Edit*: if you arrived at this issue for any reason, [make sure to read the blog post](https://blog.readthedocs.com/migrate-configuration-v2/) that contains all the important info** 

For some background here, we identified a number of issues that stem from our historical decision to provide a default configuration file and experience to projects that do not define a project configuration file. Because of this stance, we do have a number of projects that do not know about our configuration file and rely on these defaults. This applies to both new projects (for example projects using our tutorial template) and well aged projects (projects started using Sphinx 1.8.5 for example).

Deprecation of this pattern will require that _all_ projects define a Read the Docs configuration file in order to build. In most cases, a basic configuration file can be suggested to the project.

The downstream effect of requiring a configuration file is:

- We can drop support for old build images that are also used by builds without a configuration file.
- We can start to drop support for Sphinx 1.8.5, as we won't automatically opt projects into this version.

We discussed why we could provide _new_ defaults to projects, and the conclusion here was a little nuanced and opinionated in our favor. But at the center of this conversation was also that all other similar services require the user define a configuration file, so this doesn't need to be an expected pattern on our platform.

The order of operations here would be:

- https://github.com/readthedocs/readthedocs.org/issues/10348
- https://github.com/readthedocs/readthedocs.org/issues/10351
- https://github.com/readthedocs/readthedocs.org/issues/10352
- Do anything we can to get projects to add a configuration file before the cutoff date

**Comment by steveschnepp** [comment]
Would it be possible to have a default for every configuration options. Ideally set to the **_recommended_** one?

Such as `build.os` & `build.tools.python` being set to the best supported version. This would enable you to rollout new version gradually, as you will be in the driver seat. And for projects that depend on a particular version, they can opt-in to it.

Therefore if I'm just fine with every default, I'd have a file with just `version` inside. (Even that could have a default, currently `2`)

**Comment by steveschnepp** [comment]
Thanks!

Actually it was that blog post that led me to here, as I wanted to open a new issue about the defaults, but since I found this one, I simply added a comment. 😉 

I understand that the current defaults are old. But let's simply say : **_those are the `v1` defaults._** 

On the other hand, the `v2` defaults should be evolving over time, and indeed they might break many projects. 
Yet it's by design, therefore it shall be properly documented. Those projects can then opt-in to fixed versions if they need to.

Which is a part of why requiring to have an explicit config file is a good idea.
